src/tests/conformance/query/peer_shard_info.rs

```diff
@@ -24,3 +35,88 @@ async fn c006_TM_GET_PEER_SHARD_INFO_V2_node_should_not_query_for_shard_info_if_
         |m: &BinaryMessage| matches!(&m.payload, Payload::TmGetPeerShardInfoV2(..));
     perform_response_test(TestConfig::default().with_handshake(false), &response_check).await;
 }
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c011_TM_GET_PEER_SHARD_INFO_V2_node_should_relay_shard_info() {
+    // ZG-CONFORMANCE-011
+    for key_type in KEY_TYPES {
+        check_relay_for_key_type(*key_type, 2).await;
+    }
+}
+
+#[tokio::test]
+#[should_panic]
+#[allow(non_snake_case)]
+async fn c012_TM_GET_PEER_SHARD_INFO_V2_node_should_not_relay_shard_info_with_invalid_key_type() {
+    // ZG-CONFORMANCE-012
+    const INVALID_KEY: u8 = 0x42;
+    check_relay_for_key_type(INVALID_KEY, 2).await;
+}
+
+#[tokio::test]
+#[should_panic]
+#[allow(non_snake_case)]
+async fn c013_TM_GET_PEER_SHARD_INFO_V2_node_should_not_relay_shard_info_when_relays_equals_zero() {
+    // ZG-CONFORMANCE-013
+    check_relay_for_key_type(KEY_TYPES[0], 0).await;
+}
+
+#[tokio::test]
+#[should_panic]
+#[allow(non_snake_case)]
+async fn c014_TM_GET_PEER_SHARD_INFO_V2_node_should_not_relay_shard_info_when_relays_above_limit() {
+    // ZG-CONFORMANCE-014
+    const RELAY_LIMIT: u32 = 3;
+    check_relay_for_key_type(KEY_TYPES[0], RELAY_LIMIT + 1).await;
+}
+
+async fn check_relay_for_key_type(key_type: u8, relays: u32) {
+    // Create node.
+    let target = TempDir::new().expect("unable to create TempDir");
+    let mut node = Node::builder()
+        .log_to_stdout(false)
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .expect("unable to start rippled node");
```

Comment:
```suggestion
        .expect("unable to start the rippled node");
```
this is a common grammatical error in the other tests as well - we should clean it up later on

Reply:
Done